docs/vocabulary.md

```diff
@@ -416,6 +416,27 @@ In the lists and table below, Software Artifacts includes: Documentation Source
+* Aliases: Introduce
+* Inputs: Public/Upstream URL to source code repository of the primary OSS, version of the primary OSS, Organizational Policies
+* Outputs: Metadata, records, logs, SBOM, evaluation result/score based on organizational policies
+* Other Results and Side Effects: The Introducation stage may also download community generated SBOM for the given version of the OSS if available, generate SBOM using the tools organization uses, generate statistics for determining community health (e.g., no of contributors, release frequency), analyze the primary OSS to identify its dependencies, identify license/copyright information, analyze software for vulnerabilities and malware, perform basic quality analysis. One or more of these outputs may contribute to final evaluation result/score and may result in blocking the introduction of the primary OSS.
+
+#### OSS Storage Stage
+* Semantics: Download the source code of the primary OSS from its origin, analyze it for its dependencies, download the source code of the dependencies from their origin, and store them on internal/secure/trusted repositories for further activities as well as for making it available for consumption by the rest of the development organization.
+* Aliases: Store
+* Inputs: Public/Upstream URL to source code repository of the primary OSS, version of the primary OSS, Organizational Policies
+* Outputs: Metadata, records, reports, logs, SBOM, source code of the primary OSS on internal/secure/trusted repository, source code of the dependencies on internal/secure/trusted repository, evaluation result/score based on organizational policies
+* Other Results and Side Effects: The Storage stage may also generate delta report for the dependencies of the primary OSS using the previously introduced version of the same OSS and the curent one. Additionally, this stage may also contribute to evaluation result/score based on the dependencies that go through OSS Introduction stage individually and if they are not adherent to organizational policies due to various reasons such as copyright/license and community health.
+
+#### OSS Consumption Stage
+* Semantics: Build primary OSS and its dependencies using their source code stored on internal/secure/trusted repositories on a secure build environment and store resulting signed artifacts/binaries on internal/secure/trusted repositories.
+* Aliases: Consume
+* Inputs: Internal URL to source code repository of the primary OSS, version of the primary OSS, Internal URL to source code repository of the dependencies
+* Outputs: Metadata, records, reports, logs, build BOM, signed artifacts/binaries
+* Other Results and Side Effects: The Consumption stage may also generate delta report using the build BOM of the previously build artifact and the build BOM of the current artifact.
+
```

Comment:
Can you add these to the "Inputs and Outputs for Pipeline Stages" table as well?

Reply:
Why did you decide to make these 3 Stages instead of Steps?  Is it because they're usually done outside of the build process?

Reply:
> Can you add these to the "Inputs and Outputs for Pipeline Stages" table as well? [blocking]

Will add them once we have some more discussion around if the stages proposed here makes sense and once we reach a consensus.

> Why did you decide to make these 3 Stages instead of Steps? Is it because they're usually done outside of the build process?

I think there are multiple steps one needs to perform while bringing new OSSes.
For example the FOSS Introduction stage could have steps performing

- License/copyright analysis/identification
- Community/project assessment (e.g., health, release rate)
- Basic quality checks (e.g., static analysis)

Some of the steps included in the original document such as build and dependency discovery could be reused by the stages documented in this PR.

The list of possible steps is documented on HackMD but not reflected to this document yet:
- [OSS Introduction](https://hackmd.io/U6q685gFTdWRrkWZechvGw?view#OSS-Introduction)
- [OSS Storage](https://hackmd.io/U6q685gFTdWRrkWZechvGw?view#OSS-Storage)
- [OSS Consumption](https://hackmd.io/U6q685gFTdWRrkWZechvGw?view#OSS-Consumption)

Reply:
But then they should be steps and not stages, or did I misinterpret you?

Reply:
I think I'm confused by the "Storage" and "Consumption" language -- "Storage" means to me "mirror" / "vendor" and "consumption" means "incorporate into my build".

My 2c: consider combining these? Or perhaps it's just an unsolved naming issue?